packages/eslint-plugin/src/rules/confusing-non-null-assertion-like-not-equal.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  AST_NODE_TYPES,
+  AST_TOKEN_TYPES,
+  TSESLint,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+import * as util from '../util';
+
+type MessageIds = 'notNeedInEqualTest' | 'confusing';
+
+export default util.createRule<[], MessageIds>({
+  name: 'confusing-non-null-assertion-like-not-equal',
+  meta: {
+    type: 'problem',
+    docs: {
+      description:
+        'Disallows confusing combinations of non-null assertion and equal test like `a! == b`, which looks very similar to not equal `a !== b`',
+      category: 'Stylistic Issues',
+      recommended: 'error',
```

Comment:
Cannot add rules to recommended without a breaking release

```suggestion
      recommended: false,
```